torch/csrc/jit/script/compiler.cpp

```diff
@@ -1037,10 +1041,7 @@ struct to_ir {
   }
 
   Value* emitListComprehension(const ListComp& lc) {
-    // this avoids a race condition where we would re-use the same temp name
-    static std::atomic<size_t> tmp_count{0};
-    const auto tmp_name =
-        std::string("___list_acc") + std::to_string(tmp_count++);
+    const auto tmp_name = createTempName("___list_acc");
```

Comment:
unrelated to your diff but it's better practice to prefix the names with `$` so that there is no possibility of user collision